options/posix/generic/net-if-stubs.cpp

```diff
@@ -1,24 +1,49 @@
-
-#include <dirent.h>
+#include <errno.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
 #include <net/if.h>
 #include <bits/ensure.h>
 
-void if_freenameindex(struct if_nameindex *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+void if_freenameindex(struct if_nameindex *idx) {
+	free(idx);
 }
 
-char *if_indextoname(unsigned int, char *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+char *if_indextoname(unsigned int index, char *name) {
+	struct ifreq ifr;
+	int fd, r;
+
+	if((fd = socket(AF_UNIX, SOCK_DGRAM|SOCK_CLOEXEC, 0)) < 0) {
+		return 0;
+	}
+
+	ifr.ifr_ifindex = index;
+	r = ioctl(fd, SIOCGIFNAME, &ifr);
+	close(fd);
+	if (r < 0) {
+		if(errno == ENODEV) {
+			errno = ENXIO;
+		}
+		return 0;
+	}
+	return strncpy(name, ifr.ifr_name, IF_NAMESIZE);
 }
 
 struct if_nameindex *if_nameindex(void) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+	errno = ENOSYS;
+	return NULL;
 }
 
-unsigned int if_nametoindex(const char *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+unsigned int if_nametoindex(const char *name) {
+	struct ifreq ifr;
+	int fd, r;
+
+	if((fd = socket(AF_UNIX, SOCK_DGRAM|SOCK_CLOEXEC, 0)) < 0) {
+		return 0;
+	}
+	strncpy(ifr.ifr_name, name, sizeof ifr.ifr_name);
```

Comment:
function style sizeof

Reply:
Are we sure that the strncpy call behaves correctly here? e.g if `name` is too long, then `ifr.ifr_name` won't be null terminated.